src/main/java/seedu/address/model/person/Fee.java

```diff
@@ -0,0 +1,83 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's outstanding fees in TAB.
+ * Guarantees: immutable; is valid as declared in {@link #isValidFee(String)}
+ */
+public class Fee {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Fees should be formatted with a decimal point '.' as a separator between the dollars and cents, "
+            + "and adhere to the following constraints:\n"
+            + "1. Fees should only contain numbers and at most one decimal point.\n"
+            + "2. Fees should not start or end with a decimal point and should have at most two decimal places.";
+
+    public static final String VALIDATION_REGEX = "^[0-9]+(\\.[0-9]{1,2})?$";
+    public final String value;
+
+    /**
+     * Constructs an outstanding {@code Fee}.
+     *
+     * @param fee A valid outstanding fee.
+     */
+    public Fee(String fee) {
+        requireNonNull(fee);
+        checkArgument(isValidFee(fee), MESSAGE_CONSTRAINTS);
+        value = friendlyFee(fee);
+    }
+
+    /**
+     * Removes leading zeroes and postfixes second decimal place.
+     *
+     * @param fee A valid fee.
+     * @return The user-friendly fee.
+     */
+    private String friendlyFee(String fee) {
+        String friendlyFee = fee;
+        if (friendlyFee.startsWith("0")) { // remove all leading zeroes
+            friendlyFee = friendlyFee.replaceFirst("^0+", "");
+        }
+        if (friendlyFee.startsWith(".")) { // prefix missing zero
+            friendlyFee = "0" + friendlyFee;
+        }
+        int length = fee.length();
+        if (length >= 2 && fee.charAt(length - 2) == '.') { // postfix missing zero
+            friendlyFee = friendlyFee + "0";
+        }
+        if (friendlyFee.equals("0.00")) { // ensure only one representation of zero
+            friendlyFee = "0";
+        }
+        return fillEmptyString(friendlyFee);
```

Comment:
Maybe you can try using `NumberFormat` and BigDecimal to save you the trouble to manually checking and converting to a standard currency format? 
```
String a = ".1";
String b = "1000";

// USD $ and SGD $ look the same so I used Locale.US
NumberFormat nf = NumberFormat.getCurrencyInstance(Locale.US); 

System.out.println(nf.format(new BigDecimal(a))); // prints $0.10; 
System.out.println(nf.format(new BigDecimal(b))); // prints $1,000.00
```

but the standard `0` for this is `$0.00`.

Reply:
oooh thanks for the suggestion i didn't know about this, ill check it out

Reply:
@Xiaoyunnn I encountered a lot of problems when changing to this, and the new solution required just as much checking and more exception handling. Since I feel that the parsing is good enough as it is, I will not be implementing it this way.